lunatrace/bsl/backend/src/models/vulnerability-dependency-tree/index.ts

```diff
@@ -141,7 +143,12 @@ export default class VulnerabilityDependencyTree {
   public getVulnerableReleases(): VulnerableRelease[] {
     const vulnerableReleasesById: Map<string, VulnerableRelease> = new Map();
 
-    this.vulnerableNodes.forEach((nodeId) => {
+    const previouslyLoopedNodeIds: string[] = [];
+    this.vulnerableNodeIds.forEach((nodeId) => {
+      if (previouslyLoopedNodeIds.includes(nodeId)) {
+        throw new Error('dupe node looped');
```

Comment:
either remove this or improve the message that is thrown and in the places that `getVulnerableReleases` is called, wrap in a try catch.